### PR TITLE
add option to provide a protoc definition

### PR DESCRIPTION
**Pull request by kissmikijr**
We'd love to be able to use this library in a way where we fetch the content of a protoc generated json first and then just feed the data into this react component. 
We need to do this in cases where you want to host your generated json files behind some kind of authentication.

This PR adds the optional parameter `definition` to the component so you can skip the fetching and it will just render the definition that you provide. It should be a valid JSON string.

I made the `file` parameter optional.

### CODE REVIEW

**Comment by kissmikijr** [comment]
@bartek1912 Could you check this PR out ?

File: src/components/widgets/pseudomutoProtocGenDoc.tsx [hunk 1]
```diff
@@ -51,122 +53,130 @@ export const PseudomutoProtocGenDoc = ({
...
+      }
+      let protocDefinition
+      if (definition) {
+        protocDefinition = JSON.parse(definition)
+      } else if (file) {
+        const resp = await fetch(file)
+        protocDefinition = await resp.json()
+      }
+
+      if (!file && !definition) {
```
Comment (bartek1912) [inline src/components/widgets/pseudomutoProtocGenDoc.tsx:70]: Can you also throw an error when both are defined?

File: example/src/App.tsx [hunk 2]
```diff
@@ -6,951 +6,10 @@ import { GRPCSelfGeneratedAPIReference } from 'grpc-docs'
 const App = () => {
   // return <GRPCGenDocuAPIReference project='LibraryApp' organization='gendocu' />
   // return <GRPCGenDocuAPIReference project='GendocuPublicApis' organization='gendocu' />
+
   return (
-    <GRPCSelfGeneratedAPIReference
```
Comment (bartek1912) [inline example/src/App.tsx]: Having an example of the `definition` param would be great. Can you add a commented-out line with the example of usage? And please extract the definition value to some constant in another file in `example/src/`.

**Review by bartek1912 — CHANGES_REQUESTED** [review]
Looks great! Can you also drop a few words about this new parameter in the readme?

**Review by bartek1912 — APPROVED** [review]

**Comment by kissmikijr** [comment]
Thanks! I can not merge it :(
One question, what is the release process ? Should I bump the package.json ?

**Comment by bartek1912** [comment]
I've released it with 1.1.1 version ;)

**Pull request merged**